### PR TITLE
Switch our CoreCLR projects to be ".NETPortable,v5.0"

### PR DESCRIPTION
**Pull request by Pilchie**
This fixes an issue in VB where we would add the desktop mscorlib to the
project.

### CODE REVIEW

**Comment by Pilchie** [comment]
Tagging @agocke, @tmat, @davkean, @rchande, @jasonmalinowski

File: src/Compilers/CSharp/CscCore/CscCore.csproj [hunk 1]
```diff
@@ -18,8 +18,10 @@
     <OutDir>$(OutDir)core-clr\</OutDir>
     <RestorePackages>true</RestorePackages>
     <AutoGenerateBindingRedirects>True</AutoGenerateBindingRedirects>
-    <TargetFrameworkIdentifier>DNXCore</TargetFrameworkIdentifier>
+    <ProjectTypeGuids>{786C830F-07A1-408B-BD7F-6EE04809D6DB};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
+    <TargetFrameworkIdentifier>.NETPortable</TargetFrameworkIdentifier>
     <TargetFrameworkVersion>v5.0</TargetFrameworkVersion>
+
     <!-- Misspelling, bug in NuGet targets -->
```
Comment (jasonmalinowski) [inline src/Compilers/CSharp/CscCore/CscCore.csproj]: Can we move this hack into one place?

File: build/Targets/VSL.Imports.targets [hunk 2]
```diff
@@ -49,30 +49,6 @@
...
-  <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == 'DNXCore' and '$(OS)' != 'Windows_NT'">
-      <!-- Manually set assembly search paths so they don't get mangled later -->
-      <AssemblySearchPaths Condition=" '$(AssemblySearchPaths)' == ''">
-          {RawFileName};
-          $(OutDir)
-      </AssemblySearchPaths>
-      <RemoveAssemblyFoldersIfNoTargetFramework>false</RemoveAssemblyFoldersIfNoTargetFramework>
-  </PropertyGroup>
-
-  <!-- Work around issue in dotnet/roslyn issue #5213
```
Comment (jasonmalinowski) [inline build/Targets/VSL.Imports.targets]: I assume you're closing this bug?

**Comment by jasonmalinowski** [comment]
:+1:

**Comment by Pilchie** [comment]
Well.  That's a problem. :frowning:.  Looks like mono doesn't have support for portable v5.0 yet.
```
12:41:30 /mnt/j/workspace/roslyn_prtest_lin_dbg_unit32/build/Targets/VSL.Imports.targets(54,3): error MSB4019: The imported project "/tmp/mono.linux.3/lib/mono/xbuild/Microsoft/Portable/v5.0/Microsoft.Portable.CSharp.targets" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk. [/mnt/j/workspace/roslyn_prtest_lin_dbg_unit32/src/Compilers/CSharp/CscCore/CscCore.csproj]
```

File: build/Targets/VSL.Imports.targets [hunk 3]
```diff
@@ -49,30 +49,6 @@
...
-  </PropertyGroup>
-
-  <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == 'DNXCore'">
-      <BaseNuGetRuntimeIdentifier Condition="'$(OS)' == 'Windows_NT'">win7</BaseNuGetRuntimeIdentifier>
-      <BaseNuGetRuntimeIdentifier Condition="'$(OS)' == 'Unix'">ubuntu.14.04</BaseNuGetRuntimeIdentifier>
-  </PropertyGroup>
-
   <Import Project="$(MSBuildExtensionsPath32)\Microsoft\Portable\$(TargetFrameworkVersion)\Microsoft.Portable.VisualBasic.targets"
           Condition="'$(ProjectLanguage)' == 'VB' And '$(TargetFrameworkIdentifier)' == '.NETPortable'"/>
   <Import Project="$(MSBuildExtensionsPath32)\Microsoft\Portable\$(TargetFrameworkVersion)\Microsoft.Portable.CSharp.targets"
```
Comment (Pilchie) [inline build/Targets/VSL.Imports.targets]: What do the portable targets get us in this case @rchande / @davkean?  What if I made this conditional on OS==Windows_NT?

**Comment by agocke** [comment]
@Pilchie We control the mono drop. Theoretically we could copy over the portable 5.0 targets. I think we should probably do that anyway since we'll need it when I start moving unit tests over.

**Comment by Pilchie** [comment]
@agocke How/where should this change be made (should I just let you take over here?)

**Comment by agocke** [comment]
@Pilchie I'd be happy to make the addition if I can get a pointer to what we consider the "canonical" Portable 5.0 targets directory.

**Comment by Pilchie** [comment]
`C:\Program Files (x86)\MSBuild\Microsoft\Portable\v5.0` seems promising :smile:

**Comment by borgdylan** [comment]
@Pilchie If MSFT starts distributing reference assemblies for PCL v.5 as a zip file, then they could be installed on mono using simple xcopy.

**Comment by davkean** [comment]
It's not the reference assemblies that are problem, v5.0 portable's reference assemblies come completely via NuGet. It's the target files.

**Comment by borgdylan** [comment]
By any chance are they the .NET core reference assemblies that bear the "dotnet" TFM?

**Comment by davkean** [comment]
Yes.

**Pull request merged**